app/src/main/java/com/esoxjem/movieguide/listing/MoviesListingActivity.java

```diff
@@ -48,6 +52,38 @@ private void setToolbar() {
     @Override
     public boolean onCreateOptionsMenu(Menu menu) {
         getMenuInflater().inflate(R.menu.menu_main, menu);
+        final MenuItem searchItem = menu.findItem(R.id.action_search);
+        final SearchView searchView = (SearchView) searchItem.getActionView();
+
+        searchItem.setOnActionExpandListener(new MenuItem.OnActionExpandListener() {
+            @Override
+            public boolean onMenuItemActionExpand(MenuItem item) {
+                return true;
+            }
+
+            @Override
+            public boolean onMenuItemActionCollapse(MenuItem item) {
+                MoviesListingFragment mlFragment = (MoviesListingFragment) getSupportFragmentManager().findFragmentById(R.id.fragment_listing);
+                mlFragment.searchViewBackButtonClicked();
+                return true;
+            }
+        });
+
+        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
+            @Override
+            public boolean onQueryTextSubmit(String query) {
+                SoftKeyboardUtils.hideSoftInput(searchView);
+                MoviesListingFragment mlFragment = (MoviesListingFragment) getSupportFragmentManager().findFragmentById(R.id.fragment_listing);
+                mlFragment.searchViewCliked(query);
+                return true;
+            }
+
+            @Override
+            public boolean onQueryTextChange(String newText) {
```

Comment:
What do you think if you made the search as the user typed? instead of waiting until he/she pressed the search button??

Reply:
That should be nice but We need to query tmdb database to get the results, making network call for each character user typed is too much

Reply:
@rijogeorge7 you can use the debounce operator in RxJava and set it to something like 500ms